controllers/podplacement/pod_model.go

```diff
@@ -429,3 +492,26 @@ func (pod *Pod) handleError(err error, s string) {
 	pod.publishEvent(corev1.EventTypeWarning, ImageArchitectureInspectionError, ImageArchitectureInspectionErrorMsg+err.Error())
 	log.Error(err, s)
 }
+
+func (pod *Pod) isPreferredAffinityConfiguredForArchitecture() bool {
+	if pod.Spec.Affinity == nil {
+		return false
+	}
+
+	if pod.Spec.Affinity.NodeAffinity == nil {
+		return false
+	}
+
+	if pod.Spec.Affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution == nil {
+		return false
+	}
```

Comment:
Use [short-circuit](https://kuree.gitbooks.io/the-go-programming-language-report/content/22/text.html):
```suggestion
	if pod.Spec.Affinity == nil || pod.Spec.Affinity.NodeAffinity == nil || 
		pod.Spec.Affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution == nil {
		return false
	}
```